backend/src/test/java/serviceTests/CategoryServiceTest.java

```diff
@@ -0,0 +1,395 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CategoryServiceTest {
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @InjectMocks
+    private CategoryService categoryService;
+
+
+    @Test
+    public void testGetAllCategories() {
+        List<Category> categoryList = new ArrayList<>();
+
+        Category category1 = new Category();
+        category1.setId(1L);
+        category1.setName("Category 1");
+        category1.setDescription("Description 1");
+
+        Category category2 = new Category();
+        category2.setId(2L);
+        category2.setName("Category 2");
+        category2.setDescription("Description 2");
+
+        categoryList.add(category1);
+        categoryList.add(category2);
+
+        List<CategoryDTO> expectedCategoryDTOList = new ArrayList<>();
+
+        CategoryDTO categoryDTO1 = new CategoryDTO();
+        categoryDTO1.setId(1L);
+        categoryDTO1.setName("Category 1");
+        categoryDTO1.setDescription("Description 1");
+
+        CategoryDTO categoryDTO2 = new CategoryDTO();
+        categoryDTO2.setId(2L);
+        categoryDTO2.setName("Category 2");
+        categoryDTO2.setDescription("Description 2");
+
+        expectedCategoryDTOList.add(categoryDTO1);
+        expectedCategoryDTOList.add(categoryDTO2);
+
+        Pageable pageable = PageRequest.of(0, 10);
+
+        Page<Category> categoryPage = new PageImpl<>(categoryList);
+
+        when(categoryRepository.findAll(any(Pageable.class))).thenReturn(categoryPage);
+        when(categoryMapper.toDto(any(Category.class))).thenAnswer(invocationOnMock -> {
+            Category category = invocationOnMock.getArgument(0);
+            Long id = category.getId();
+            return expectedCategoryDTOList.stream()
+                    .filter(dto -> dto.getId().equals(id))
+                    .findFirst()
+                    .orElse(null);
+        });
+
+        Page<CategoryDTO> result = categoryService.getAllCategories(pageable);
+
+        assertEquals(2, result.getTotalElements());
+        assertEquals(expectedCategoryDTOList.get(0).getId(), result.getContent().get(0).getId());
+        assertEquals(expectedCategoryDTOList.get(0).getName(), result.getContent().get(0).getName());
+        assertEquals(expectedCategoryDTOList.get(0).getDescription(), result.getContent().get(0).getDescription());
+
+        assertEquals(expectedCategoryDTOList.get(0).getId(), result.getContent().get(0).getId());
+        assertEquals(expectedCategoryDTOList.get(0).getName(), result.getContent().get(1).getName());
+        assertEquals(expectedCategoryDTOList.get(0).getDescription(), result.getContent().get(1).getDescription());
+    }
+
+    @Test
+    public void testSearchCategories() {
```

Comment:
you need to read about unit test naming, e.g. https://www.baeldung.com/java-unit-testing-best-practices